webapp/server/middlewares/auth.ts

```diff
@@ -20,32 +20,17 @@ function isProtectedRoute(req: Request) {
 export async function auth(req: Request, res: Response, next: NextFunction) {
```

Comment:
This function was re writen. The idea is that it always tries to get the address and role for the user. If an anonymous user tries to go to a protected route it redirects to denied.